crates/telio-pq/src/proto.rs

```diff
@@ -187,13 +184,13 @@ async fn handshake(
     }
 
     // The response should be 92, so the buffer is sufficient
-    let read = sock.recv(&mut pkgbuf).await?;
+    let pkg = sock.recv(&mut pkgbuf).await?;
```

Comment:
This is the place that was missing the timeout? It can be tested by starting a udp server that will ignore the handshake that is sent above and will not reply to it in any way. This way you can check in a test that after ~RECV_TIMEOUT the `async fn handshake` will return with an error.

Reply:
Yes that was the place

Reply:
Do you mean nat-lab test? The `handshake` function is not exposed

Reply:
I mean pure rust test.

Reply:
Let me think of something

Reply:
I've added tests for `fetch_keys` and `rekey` since those are the interface functions